client/app/components/tags-control/TagsControl.jsx

```diff
@@ -24,7 +24,9 @@ export class TagsControl extends React.Component {
   };
 
   editTags = (tags, getAvailableTags) => {
-    EditTagsDialog.showModal({ tags, getAvailableTags }).result.then(this.props.onEdit);
+    EditTagsDialog.showModal({ tags, getAvailableTags })
+      .result.then(this.props.onEdit)
+      .catch(() => {}); // ignore dismiss
```

Comment:
Coudln't we handle this in `showModal` instead of having to add this everywhere?

Reply:
Sadly, no :slightly_frowning_face:  After `.catch()` handler promise will go to next `.then()` handler, so if we catch that promises inside of dialog wrapper - the caller's `.catch()` will not be called. It actually is not a problem, but browser will show unhandled rejection errors in console. We can either:
1. don't catch that promises and ignore errors in console;
2. always add empty catch handler like I did here;
3. replace browser's Promise with something else that does not trigger unhandled rejection errors (e.g. "bluebird") - prior to this PR we used Angular's `$q` for this purpose;
4. change a way we handle dialog's close/dismiss events (for me current flow looks fine, but that annoying unhandled rejections...)

Reply:
I'll explore if we can suppress that error with handling `unhandledrejection` event if error comes from dialog wrapper

Reply:
Ok, let's continue the conversation here: https://github.com/getredash/redash/issues/4566.